File: apps/android/app/src/main/java/com/litter/android/ui/common/AgentRuntimePresentation.kt
package com.litter.android.ui.common

import androidx.annotation.DrawableRes
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.litter.android.ui.LitterTheme
import com.sigkitten.litter.android.R
import uniffi.codex_mobile_client.AgentRuntimeKind

val AgentRuntimeKind.runtimeLabel: String
    get() = when (this) {
        AgentRuntimeKind.CODEX -> "Codex"
        AgentRuntimeKind.PI -> "Pi"
        AgentRuntimeKind.AMP -> "Amp"
        AgentRuntimeKind.OPENCODE -> "opencode"
        AgentRuntimeKind.CLAUDE -> "Claude"
        AgentRuntimeKind.DROID -> "Droid"
    }

@get:DrawableRes
val AgentRuntimeKind.runtimeDrawable: Int
    get() = when (this) {
        AgentRuntimeKind.CODEX -> R.drawable.agent_codex
        AgentRuntimeKind.PI -> R.drawable.agent_pi
        AgentRuntimeKind.AMP -> R.drawable.agent_amp
        AgentRuntimeKind.OPENCODE -> R.drawable.agent_opencode
        AgentRuntimeKind.CLAUDE -> R.drawable.agent_claude
        AgentRuntimeKind.DROID -> R.drawable.agent_droid
    }

val AgentRuntimeKind.runtimeSortIndex: Int
    get() = when (this) {
        AgentRuntimeKind.CODEX -> 0
        AgentRuntimeKind.PI -> 1
        AgentRuntimeKind.AMP -> 2
        AgentRuntimeKind.OPENCODE -> 3
        AgentRuntimeKind.CLAUDE -> 4
        AgentRuntimeKind.DROID -> 5
    }

val AgentRuntimeKind.isBeta: Boolean
    get() = when (this) {
        AgentRuntimeKind.CLAUDE,
        AgentRuntimeKind.PI,
        AgentRuntimeKind.AMP,
        AgentRuntimeKind.OPENCODE,
        AgentRuntimeKind.DROID -> true
        AgentRuntimeKind.CODEX -> false
    }

private val betaAgentNameAliases = setOf(
    "claude", "claude-code", "claude_code",
    "pi", "pi.dev", "pidev",
    "amp", "ampcode", "amp-code", "amp_code", "amp code",
    "opencode", "open-code", "open_code", "open code",
    "droid", "factory", "factory-droid", "factory_droid", "factory droid",
)

fun isBetaAgentName(name: String, displayName: String): Boolean =
    name.lowercase() in betaAgentNameAliases || displayName.lowercase() in betaAgentNameAliases

@Composable
fun BetaBadge(modifier: Modifier = Modifier) {
    Box(
        modifier = modifier
            .border(
                width = 0.5.dp,
                color = LitterTheme.accent.copy(alpha = 0.6f),
                shape = RoundedCornerShape(3.dp),
            )
            .padding(horizontal = 5.dp, vertical = 1.dp),
    ) {
        Text(
            text = "BETA",
            color = LitterTheme.accent,
            fontSize = 9.sp,
            fontWeight = FontWeight.SemiBold,
        )
    }
}

File: apps/ios/Sources/Litter/Views/AgentRuntimePresentation.swift
import SwiftUI

extension AgentRuntimeKind {
    static let presentationOrder: [AgentRuntimeKind] = [
        .codex,
        .pi,
        .amp,
        .opencode,
        .claude,
        .droid,
    ]

    var displayLabel: String {
        switch self {
        case .codex:
            return "Codex"
        case .pi:
            return "Pi"
        case .amp:
            return "Amp"
        case .opencode:
            return "opencode"
        case .claude:
            return "Claude"
        case .droid:
            return "Droid"
        }
    }

    var titleDisplayLabel: String {
        switch self {
        case .opencode:
            return "Opencode"
        case .droid:
            return "Factory Droid"
        case .amp:
            return "Amp"
        default:
            return displayLabel
        }
    }

    var assetName: String {
        switch self {
        case .codex:
            return "agent_codex"
        case .pi:
            return "agent_pi"
        case .amp:
            return "agent_amp"
        case .opencode:
            return "agent_opencode"
        case .claude:
            return "agent_claude"
        case .droid:
            return "agent_droid"
        }
    }

    var systemImageName: String {
        switch self {
        case .codex:
            return "terminal"
        case .pi:
            return "circle.hexagongrid"
        case .amp:
            return "bolt"
        case .opencode:
            return "chevron.left.forwardslash.chevron.right"
        case .claude:
            return "sparkle"
        case .droid:
            return "gearshape"
        }
    }

    var presentationSortIndex: Int {
        Self.presentationOrder.firstIndex(of: self) ?? Int.max
    }

    var isBeta: Bool {
        switch self {
        case .claude, .pi, .amp, .opencode, .droid:
            return true
        case .codex:
            return false
        }
    }

    static func isBetaAgentName(_ name: String, displayName: String) -> Bool {
        let normalized = name.lowercased()
        let display = displayName.lowercased()
        let aliases: Set<String> = [
            "claude", "claude-code", "claude_code",
            "pi", "pi.dev", "pidev",
            "amp", "ampcode", "amp-code", "amp_code", "amp code",
            "opencode", "open-code", "open_code", "open code",
            "droid", "factory", "factory-droid", "factory_droid", "factory droid",
        ]
        return aliases.contains(normalized) || aliases.contains(display)
    }
}

struct BetaBadge: View {
    var body: some View {
        Text("BETA")
            .litterFont(.caption2)
            .foregroundColor(LitterTheme.accent)
            .padding(.horizontal, 5)
            .padding(.vertical, 1)
            .overlay(
                RoundedRectangle(cornerRadius: 3)
                    .stroke(LitterTheme.accent.opacity(0.6), lineWidth: 0.5)
            )
    }
}
